src/extension/PostProblems.ts

```diff
@@ -0,0 +1,332 @@
+          return
+        }
+
+        const path = await this.getPathToProblem(regexPathArray, logFilePath)
+
+        const descriptiveMessage = curMessage
+          .replace(pathRegex, '')
+          .replace(locationRegex, '')
+          .replace(' ()', '') // for dealing with spec file errors messages
+
+        const position: Position = this.getPosition(
+          regexLocationArray,
+          path,
+          logFilePath,
+        )
+        /**
+         * right now we are only getting the start position from the resource_errors.json message
+         * vscode handles this range by emphasizing the words that start at position [position]
+         */
+        const diagnostic: Diagnostic = this.createDiagnostic(
+          position,
+          position,
+          descriptiveMessage,
+        )
+        this.setDiagnosticMap(diagnostic, path, diagnosticMap)
+      }
+    }
+    this.postDiagnostics(diagnosticMap)
+  }
+
+  /**
+   * returns a new diagnostic object
+   * @param startPosition position where the problem starts
+   * @param endPosition position where the problem ends
+   * @param message message describing the problem
+   * @returns diagnostic object
+   */
+  private createDiagnostic(
+    startPosition: Position,
+    endPosition: Position,
+    message: string,
+  ): Diagnostic {
+    const range: Range = new Range(startPosition, endPosition)
+    const diagnostic = new Diagnostic(range, message)
+    return diagnostic
+  }
+
+  /**
+   * sets the diagnostic map [diagnosticMap] with the new value:
+   * @param diagnostic Diagnostic object
+   * @param path path to the file where problems originated from
+   * @param diagnosticMap maps paths (string) to diagnostic lists
+   */
+  private setDiagnosticMap(
+    diagnostic: Diagnostic,
+    path: string,
+    diagnosticMap: Map<string, Diagnostic[]>,
+  ): void {
+    const diagnostics: Diagnostic[] = diagnosticMap.get(path) || []
+    diagnostics.push(diagnostic)
+    diagnosticMap.set(path, diagnostics)
+  }
+
+  /**
+   * post all diagnostic lists in diagnosticMap to [PROBLEMS]
+   * @param diagnosticMap maps paths (string) to diagnostic lists
+   */
+  private postDiagnostics(diagnosticMap: Map<string, Diagnostic[]>): void {
+    diagnosticMap.forEach((diagnosticList, path) => {
+      const curDiagnosticCollection: vscode.DiagnosticCollection =
+        vscode.languages.createDiagnosticCollection()
+      curDiagnosticCollection.set(Uri.parse(path), diagnosticList)
+      this.diagnosticCollection.push(curDiagnosticCollection)
+    })
+    // when user edits file with diagnostics - clear related diagnostics
+    if (this.diagnosticCollection.length > 0) {
+      this.deleteOnEdit()
+    }
+  }
+
+  private getResourceError(str: string): ResourceError {
+    try {
+      const jsonContent: ResourceError = JSON.parse(str)
+      return jsonContent
+    } catch (e) {
+      console.error("Couldn't read the error logs. Please contact Certora team")
```

Comment:
I didn't want it to crash, that is Why I returned an empty object.
It is a good question because if the format of the resource_errors file changes, I think this is an inner problem of certora not an outside problem, so maybe we want to keep this as an inner error message.